dbt/adapters/snowflake/connections.py

```diff
@@ -273,44 +349,8 @@ def _get_access_token(self) -> str:
             )
         return result_json["access_token"]
 
-    def _get_private_key(self):
-        """Get Snowflake private key by path, from a Base64 encoded DER bytestring or None."""
-        if self.private_key and self.private_key_path:
-            raise DbtConfigError("Cannot specify both `private_key`  and `private_key_path`")
-
-        if self.private_key_passphrase:
-            encoded_passphrase = self.private_key_passphrase.encode()
-        else:
-            encoded_passphrase = None
-
-        if self.private_key:
-            if self.private_key.startswith("-"):
-                p_key = serialization.load_pem_private_key(
-                    data=bytes(self.private_key, "utf-8"),
-                    password=encoded_passphrase,
-                    backend=default_backend(),
-                )
-
-            else:
-                p_key = serialization.load_der_private_key(
-                    data=base64.b64decode(self.private_key),
-                    password=encoded_passphrase,
-                    backend=default_backend(),
-                )
-
-        elif self.private_key_path:
-            with open(self.private_key_path, "rb") as key:
-                p_key = serialization.load_pem_private_key(
-                    key.read(), password=encoded_passphrase, backend=default_backend()
-                )
-        else:
-            return None
-
-        return p_key.private_bytes(
-            encoding=serialization.Encoding.DER,
-            format=serialization.PrivateFormat.PKCS8,
-            encryption_algorithm=serialization.NoEncryption(),
-        )
+    def _get_private_key(self) -> Optional[bytes]:
+        return _private_key(self.private_key, self.private_key_path, self.private_key_passphrase)
```

Comment:
Nit: If you want to keep this method in the class, you can use `functools.cached_property`: https://docs.python.org/3/library/functools.html#functools.cached_property

Reply:
While it's highly unlikely, if someone builds on `dbt-snowflake` and overwrites this for some reason, we'll break them. Also, putting the body of the function outside of the class means I can test it without mocking anything. I'm looking at adding a test or two tonight.